api/v1alpha1/servermaintenance_types.go

```diff
@@ -0,0 +1,84 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ServerMaintenanceReasonAnnotationKey = "metal.ironcore.dev/reason"
+)
+
+// ServerBootConfigurationTemplate defines the parameters to be used for rendering a boot configuration.
+type ServerBootConfigurationTemplate struct {
+	// Name specifies the name of the boot configuration.
+	Name string `json:"name"`
+	// Parameters specifies the parameters to be used for rendering the boot configuration.
+	Spec ServerBootConfigurationSpec `json:"spec"`
+}
+
+// ServerMaintenanceSpec defines the desired state of a ServerMaintenance
+type ServerMaintenanceSpec struct {
+	// Policy specifies the maintenance policy to be enforced on the server.
+	Policy ServerMaintenancePolicy `json:"policy,omitempty"`
+	// ServerRef is a reference to the server that is to be maintained.
+	ServerRef corev1.LocalObjectReference `json:"serverRef"`
+	// ServerPower specifies the power state of the server during maintenance.
+	ServerPower Power `json:"serverPower,omitempty"`
+	// ServerBootConfigurationTemplate specifies the boot configuration to be applied to the server during maintenance.
+	ServerBootConfigurationTemplate *ServerBootConfigurationTemplate `json:"serverBootConfigurationTemplate,omitempty"`
+}
+
+type ServerMaintenancePolicy string
+
+const (
+	ServerMaintenancePolicyOwnerApproval ServerMaintenancePolicy = "OwnerApproval"
+	ServerMaintenancePolicyEnforced      ServerMaintenancePolicy = "Enforced"
+)
+
+// ServerMaintenanceStatus defines the observed state of a ServerMaintenance
+type ServerMaintenanceStatus struct {
+	State ServerMaintenanceState `json:"state,omitempty"`
+}
+
+type ServerMaintenanceState string
+
+const (
+	ServerMaintenanceStatePending       ServerMaintenanceState = "Pending"
```

Comment:
Should there be a way to refer if the Maintenance process failed for some reason?

Reply:
Good point. Adding a `Failed` state const to the list.